fix #844 to remove all macros

sandraros opened the pull request:
fix #844 to remove all macros

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+    ENDIF.
+    IF ip_font_color IS SUPPLIED.
+      complete_style-font-color = ip_font_color.
+      complete_stylex-font-color = 'X'. " <============================== BUG ?
```
Indeed, with "color" being a zexcel_s_cstylex_color substructure that doesn't make much sense.
If I'm not mistaken, in terms of actual results it is probably equivalent to 
complete_stylex-font-color-rgb = 'X'
but we have no way, actually we never had, a way to encode that a whole "color" field was passed, regardless of which subfields were actually filled.

sandraros replied:
I guess the author wanted to set all 4 subfields to 'X', but I didn't spend time on finding if setting only `rgb` subfield is a bug or not. For now, I propose to keep the same behavior as before, by adding subfield mention "`-rgb`" so that it removes the syntax check warning. Maybe one day a bug will be filed, and we'll analyze at that time.

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+    ENDIF.
+    IF ip_fill_fgcolor IS SUPPLIED.
+      complete_style-fill-fgcolor = ip_fill_fgcolor.
+      complete_stylex-fill-fgcolor = 'X'. " <============================== BUG ?
```
See comment above for 1813

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+
+    IF ip_fill_bgcolor IS SUPPLIED.
+      complete_style-fill-bgcolor = ip_fill_bgcolor.
+      complete_stylex-fill-bgcolor = 'X'. " <============================== BUG ?
```
See comment above for 1813

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+    ENDIF.
+    IF ip_borders_allborders_color IS SUPPLIED.
+      complete_style-borders-allborders-border_color = ip_borders_allborders_color.
+      complete_stylex-borders-allborders-border_color = 'X'. " <============================== BUG ?
```
See comment above for 1813

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+    ENDIF.
+    IF ip_borders_diagonal_color IS SUPPLIED.
+      complete_style-borders-diagonal-border_color = ip_borders_diagonal_color.
+      complete_stylex-borders-diagonal-border_color = 'X'. " <============================== BUG ?
```
See comment above for 1813

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+    ENDIF.
+    IF ip_borders_down_color IS SUPPLIED.
+      complete_style-borders-down-border_color = ip_borders_down_color.
+      complete_stylex-borders-down-border_color = 'X'. " <============================== BUG ?
```
See comment above for 1813

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+    ENDIF.
+    IF ip_borders_left_color IS SUPPLIED.
+      complete_style-borders-left-border_color = ip_borders_left_color.
+      complete_stylex-borders-left-border_color = 'X'. " <============================== BUG ?
```
See comment above for 1813

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+    ENDIF.
+    IF ip_borders_right_color IS SUPPLIED.
+      complete_style-borders-right-border_color = ip_borders_right_color.
+      complete_stylex-borders-right-border_color = 'X'. " <============================== BUG ?
```
See comment above for 1813

AndreaBorgia-Abo commented on src/zcl_excel_worksheet.clas.abap:
```diff
@@ -1696,113 +1722,420 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
+    ENDIF.
+    IF ip_borders_top_color IS SUPPLIED.
+      complete_style-borders-top-border_color = ip_borders_top_color.
+      complete_stylex-borders-top-border_color = 'X'. " <============================== BUG ?
```
See comment above for 1813

AndreaBorgia-Abo reviewed (changes requested):
Other than a few minor issues with mixed-case, this is good.
The "color" flag issue was already present: as long as we preserve the (possibly broken) logic it shouldln't matter.

larshp commented:
in `abaplint.json`, suggest changing
```
    "avoid_use": {
      "define": false,
```
to
```
    "avoid_use": {
      "define": true,
```
that way all future pull request will be rejected if they contain macros/define

sandraros commented:
I did the changes as you suggested. Could you review again? Thanks!
NB: I probably should not have clicked "resolve conversation" for all your suggestions, and left it up to you. It's just that I'm discovering the process so not sure of what I'm doing. Note that you can also "suggest specific changes to lines of code, which the author can apply directly", maybe that will help make the review and corrections smoother.

sandraros commented:
Two more things that I just saw and fixed:
1. A bug I introduced, that I could reproduce with this code in case 1 (borders were shown):
    ```
    REPORT.
    DATA: gc_save_file_name TYPE string VALUE 'test.xlsx'.
    INCLUDE zdemo_excel_outputopt_incl.
    START-OF-SELECTION.
      TRY.
          DATA(lo_excel) = NEW zcl_excel( ).
          DATA(lo_worksheet) = lo_excel->get_active_worksheet( ).
          lo_worksheet->set_cell( ip_column = 'B' ip_row = 2 ip_value = CONV decfloat16( 1 ) ).
          " 1) cell should have no border
          lo_worksheet->change_cell_style(
                ip_column             = 'B'
                ip_row                = 2
                ip_borders_allborders = VALUE #( border_style = zcl_excel_style_border=>c_border_thick
                                                 border_color = VALUE #( rgb = zcl_excel_style_color=>c_darkred ) )
                ip_xborders_allborders = VALUE #( border_style = abap_false
                                                  border_color = VALUE #( rgb = abap_false ) )
                ip_xborders = VALUE #( allborders = VALUE #( border_style = abap_true
                                                             border_color = VALUE #( rgb = abap_true ) ) ) ).
          " 2) cell should have borders
          lo_worksheet->change_cell_style(
                ip_column             = 'D'
                ip_row                = 2
                ip_borders_allborders = VALUE #( border_style = zcl_excel_style_border=>c_border_thick
                                                 border_color = VALUE #( rgb = zcl_excel_style_color=>c_darkred ) )
                ip_xborders_allborders = VALUE #( border_style = abap_true
                                                  border_color = VALUE #( rgb = abap_true ) ) ).
          " 3) cell should have borders
          lo_worksheet->change_cell_style(
                ip_column             = 'F'
                ip_row                = 2
                ip_borders_allborders = VALUE #( border_style = zcl_excel_style_border=>c_border_thick
                                                 border_color = VALUE #( rgb = zcl_excel_style_color=>c_darkred ) ) ).
          lcl_output=>output( cl_excel = lo_excel ).
        CATCH cx_root INTO DATA(lx_root).
          MESSAGE lx_root TYPE 'I' DISPLAY LIKE 'E'.
      ENDTRY.
    ```
1. I also removed the changing parameter `cs_borderx` from method `move_supplied_borders`, which was useless.

AndreaBorgia-Abo commented:
> Note that you can also "suggest specific changes to lines of code, which the author can apply directly", maybe that will help make the review and corrections smoother.

Ha, I didn't know that, ["insert a suggestion"](https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request) seems useful, I'll try to use it, thanks.

Let me check the new changes...

AndreaBorgia-Abo reviewed (approved):
LGTM

AndreaBorgia-Abo commented:
@larshp I wanted to check which bug @sandraros introduced and then fixed but I couldn't find a way to check out a specific commit, only switching branches and tags are supported if I'm not mistaken. I did the next best thing, confirming the problem is indeed gone.

larshp commented:
this part should help checking out a specific commit, 
![image](https://user-images.githubusercontent.com/5888506/139707531-9db48dfd-bc02-4e10-9fb0-946cf3873b16.png)

AndreaBorgia-Abo commented:
Gosh, I kept looking in the menus... :D
Thanks.

AndreaBorgia-Abo commented:
> this part should help checking out a specific commit,
Close but no cigar: "Cannot find initial commit. Too many commits. Action not possible. "
Selection by PR works, though I'm not sure if it is any different than switching to the PR branch.
